swift/apple/FirezoneNetworkExtension/SessionManager.swift

```diff
@@ -0,0 +1,250 @@
+import FirezoneKit
+import Foundation
+import NetworkExtension
+
+/// SessionManager owns the Session lifecycle and manages command/event multiplexing.
+///
+/// This actor follows Android's pattern where the event loop owns the Session instance.
+/// When the event loop exits, the Session is automatically cleaned up, ensuring Drop
+/// is called on the Rust side to prevent memory leaks.
+///
+/// Commands are sent via AsyncStream (similar to Kotlin's Channel).
+/// Events are polled from Rust's nextEvent() and forwarded to the event handler.
+actor SessionManager {
+  // Session ownership
+  private var session: Session?
+  private var eventLoopTask: Task<Void, Never>?
+
+  // Command channel (Swift equivalent of Kotlin's Channel)
+  private let commandContinuation: AsyncStream<SessionCommand>.Continuation
+  private let commandStream: AsyncStream<SessionCommand>
+
+  // Event handler callback
+  private let eventHandler: (Event) async -> Void
+
+  // Logging interval for event loop health monitoring
+  private static let eventPollingLogInterval = 100
+
+  init(eventHandler: @escaping (Event) async -> Void) {
+    self.eventHandler = eventHandler
+    (commandStream, commandContinuation) = AsyncStream.makeStream()
+  }
+
+  /// Starts the session and event loop.
+  ///
+  /// This creates a Rust Session via UniFFI and spawns the event loop task
+  /// that owns the session lifecycle.
+  func start(
+    apiUrl: String,
+    token: Token,
+    deviceId: String,
+    accountSlug: String,
+    logFilter: String
+  ) throws {
+    Log.log("SessionManager: Starting session for account: \(accountSlug)")
+
+    // Get device metadata
+    let deviceName = DeviceMetadata.getDeviceName()
+    let osVersion = DeviceMetadata.getOSVersion()
+    let deviceInfo = try JSONEncoder().encode(DeviceMetadata.deviceInfo())
+    let deviceInfoStr = String(data: deviceInfo, encoding: .utf8) ?? "{}"
+    let logDir = SharedAccess.connlibLogFolderURL?.path ?? "/tmp/firezone"
+
+    // Create the session
+    let session: Session
+    do {
+      session = try Session.newApple(
+        apiUrl: apiUrl,
+        token: token.description,
+        deviceId: deviceId,
+        accountSlug: accountSlug,
+        deviceName: deviceName,
+        osVersion: osVersion,
+        logDir: logDir,
+        logFilter: logFilter,
+        deviceInfo: deviceInfoStr
+      )
+    } catch {
+      Log.error(error)
+      throw AdapterError.connlibConnectError(String(describing: error))
+    }
+
+    self.session = session
```

Comment:
I am not sure I understand why are saving the session here. The point of the event-loop is to own the session variable, right? Why do we need another stateful object (the `SessionManager`) when all we want to do is spawn an async function (the event-loop) as a `Task`?

It seems to me like this `SessionManager` could be folded into the adapter where the other ends of the channel are owned by the adapter and we have a free-function that acts as the event-loop.